### PR TITLE
Fix keep-alive config not working

### PR DESCRIPTION
**Pull request by riyafa**
## Automation tests
 - Unit tests 
   > Added in transport-http

### CODE REVIEW

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/config/ChunkKeepAliveConfig.java [hunk 1]
```diff
@@ -22,7 +22,7 @@
 /**
  * Contains value for configuring content encoding
  */
-public enum ChunkConfig {
+public enum ChunkKeepAliveConfig {
```
Comment (pubudu91) [inline components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/config/ChunkKeepAliveConfig.java]: Why was this renamed? The `ChunkConfig` is for configuring chunking. Keep-alive configurations shouldn't be mixed in with it IMO

Comment (riyafa) [reply components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/config/ChunkKeepAliveConfig.java]: I am using the same enum for both cases. Otherwise need to write the same enum with a different name.

**Comment by pubudu91** [comment]
@riyafa If using the same enum for both, then let's make its name a generic one. `ChunkKeepAliveConfig` just sounds weird. Also, can you explain how the keep-alive configuration map to these enum values? In what scenario does `AUTO` apply to keep-alive config?

**Comment by riyafa** [comment]
`AUTO` is for the default value, `ALWAYS` is to enable keep-alive and `NEVER` is to disable keep-alive

**Comment by pubudu91** [comment]
@riyafa But what is the default value? keep-alive is a boolean value. You either keep the connection or you close it.

**Comment by riyafa** [comment]
Default value is different for HTTP 1.0 and 1.1. 
Default for HTTP 1.0 is `close` and for HTTP 1.1 is `keep-alive`

File: components/org.wso2.transport.http.netty/src/test/java/org/wso2/transport/http/netty/http1point0test/HttpOnePointZeroServerConnectorTestCase.java [hunk 2]
```diff
@@ -83,6 +80,7 @@ public void setUp() {
     @Test
     public void http1point0DefaultRequest() {
         try {
+            setUp();
```
Comment (shafreenAnfar) [inline components/org.wso2.transport.http.netty/src/test/java/org/wso2/transport/http/netty/http1point0test/HttpOnePointZeroServerConnectorTestCase.java]: Can't we use @BeforeMethod ?

File: components/org.wso2.transport.http.netty/src/test/java/org/wso2/transport/http/netty/http1point0test/HttpOnePointZeroServerConnectorTestCase.java [hunk 3]
```diff
@@ -92,32 +90,30 @@ public void http1point0DefaultRequest() {
...
             assertNotNull(httpResponse.headers().get(Constants.HTTP_CONTENT_LENGTH));
+            serverConnector.stop();
         } catch (Exception e) {
-            TestUtil.handleException("IOException occurred while running largeHeaderTest", e);
+            TestUtil.handleException("IOException occurred while running http1point0KeepAliveRequest test", e);
         }
     }
-
-    @AfterClass
-    public void cleanUp() throws ServerConnectorException {
```
Comment (shafreenAnfar) [inline components/org.wso2.transport.http.netty/src/test/java/org/wso2/transport/http/netty/http1point0test/HttpOnePointZeroServerConnectorTestCase.java]: Here also cant we use @Aftermethod ?

Comment (shafreenAnfar) [reply components/org.wso2.transport.http.netty/src/test/java/org/wso2/transport/http/netty/http1point0test/HttpOnePointZeroServerConnectorTestCase.java]: Also we need to shutdown httpWsConnectorFactory as well..

File: components/org.wso2.transport.http.netty/src/test/java/org/wso2/transport/http/netty/http1point0test/HttpOnePointZeroServerConnectorTestCase.java (hunk 2 above)
Comment (riyafa) [reply components/org.wso2.transport.http.netty/src/test/java/org/wso2/transport/http/netty/http1point0test/HttpOnePointZeroServerConnectorTestCase.java]: We cannot in this test because we need to have different `listenerConfiguration`s for each test case.

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/listener/SourceHandler.java [hunk 4]
```diff
@@ -180,8 +184,9 @@ void notifyRequestListener(HTTPCarbonMessage httpRequestMsg, ChannelHandlerConte
             try {
                 ServerConnectorFuture outboundRespFuture = httpRequestMsg.getHttpResponseFuture();
                 outboundRespFuture
-                        .setHttpConnectorListener(new HttpOutboundRespListener(ctx, httpRequestMsg,
-                                chunkConfig, serverName));
+                        .setHttpConnectorListener(
+                                new HttpOutboundRespListener(ctx, httpRequestMsg, chunkConfig, keepAliveConfig,
+                                                             serverName));
```
Comment (anupama-pathirage) [inline components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/listener/SourceHandler.java]: Shall we reformat this line ?

File: components/org.wso2.transport.http.netty/src/test/java/org/wso2/transport/http/netty/http1point0test/HttpOnePointZeroServerConnectorTestCase.java (hunk 3 above)
Comment (riyafa) [reply components/org.wso2.transport.http.netty/src/test/java/org/wso2/transport/http/netty/http1point0test/HttpOnePointZeroServerConnectorTestCase.java]: `httpWsConnectorFactory` doesn't have a method to shut it down

Comment (riyafa) [reply components/org.wso2.transport.http.netty/src/test/java/org/wso2/transport/http/netty/http1point0test/HttpOnePointZeroServerConnectorTestCase.java]: Fixed
